src/Jackett.Common/Utils/MagnetUtil.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Text;
+using Jackett.Common.Helpers;
+
+namespace Jackett.Common.Utils
+{
+    public static class MagnetUtil
+    {
+        // Update best trackers from https://github.com/ngosang/trackerslist
+        private static readonly NameValueCollection _Trackers = new NameValueCollection
+        {
+            {"tr", "udp://tracker.opentrackr.org:1337/announce"},
+            {"tr", "udp://tracker.coppersurfer.tk:6969/announce"},
+            {"tr", "udp://9.rarbg.to:2710/announce"},
+            {"tr", "udp://tracker.internetwarriors.net:1337/announce"},
+            {"tr", "udp://tracker.cyberia.is:6969/announce"},
+            {"tr", "udp://exodus.desync.com:6969/announce"},
+            {"tr", "udp://explodie.org:6969/announce"},
+            {"tr", "http://tracker1.itzmx.com:8080/announce"},
+            {"tr", "udp://tracker.tiny-vps.com:6969/announce"},
+            {"tr", "udp://open.stealth.si:80/announce"}
+        };
+
+        private static readonly string _TrackersEncoded = _Trackers.GetQueryString();
+
+        public static Uri InfoHashToPublicMagnet(string infoHash, string title)
+        {
+            if (string.IsNullOrWhiteSpace(infoHash) || string.IsNullOrWhiteSpace(title))
+                return null;
+            return new Uri($"magnet:?xt=urn:btih:{infoHash}&dn={WebUtilityHelpers.UrlEncode(title, Encoding.UTF8)}&{_TrackersEncoded}");
+        }
```

Comment:
Hang on, the URI generated by this util is not the commonly seen version

```
 magnet:?xt=urn:btih:5c12592cee7a91552eb87f9181d3e95a3714117b&dn=SexMex.20.11.03.Pecosa.Fiery.Aunt.XXX.1080p.MP4-XXX%5BXvX%5D&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337%2Fannounce%2Cudp%3A%2F%2Ftracker.coppersurfer.tk%3A6969%2Fannounce%2Cudp%3A%2F%2F9.rarbg.to%3A2710%2Fannounce%2Cudp%3A%2F%2Ftracker.internetwarriors.net%3A1337%2Fannounce%2Cudp%3A%2F%2Ftracker.cyberia.is%3A6969%2Fannounce%2Cudp%3A%2F%2Fexodus.desync.com%3A6969%2Fannounce%2Cudp%3A%2F%2Fexplodie.org%3A6969%2Fannounce%2Chttp%3A%2F%2Ftracker1.itzmx.com%3A8080%2Fannounce%2Cudp%3A%2F%2Ftracker.tiny-vps.com%3A6969%2Fannounce%2Cudp%3A%2F%2Fopen.stealth.si%3A80%2Fannounce
```

which once decoded looks like
```
magnet:?xt=urn:btih:5c12592cee7a91552eb87f9181d3e95a3714117b&dn=SexMex.20.11.03.Pecosa.Fiery.Aunt.XXX.1080p.MP4-XXX[XvX]&tr=udp://tracker.opentrackr.org:1337/announce,udp://tracker.coppersurfer.tk:6969/announce,udp://9.rarbg.to:2710/announce,udp://tracker.internetwarriors.net:1337/announce,udp://tracker.cyberia.is:6969/announce,udp://exodus.desync.com:6969/announce,udp://explodie.org:6969/announce,http://tracker1.itzmx.com:8080/announce,udp://tracker.tiny-vps.com:6969/announce,udp://open.stealth.si:80/announce
```

the trackers are usually separated by their own &tr= rather than a single comma delimited list
```
magnet:?xt=urn:btih:5c12592cee7a91552eb87f9181d3e95a3714117b&dn=SexMex.20.11.03.Pecosa.Fiery.Aunt.XXX.1080p.MP4-XXX[XvX]&tr=udp://tracker.opentrackr.org:1337/announce,&tr=udp://tracker.coppersurfer.tk:6969/announce,&tr=udp://9.rarbg.to:2710/announce,&tr=udp://tracker.internetwarriors.net:1337/announce,&tr=udp://tracker.cyberia.is:6969/announce,&tr=udp://exodus.desync.com:6969/announce,&tr=udp://explodie.org:6969/announce,&tr=http://tracker1.itzmx.com:8080/announce,&tr=udp://tracker.tiny-vps.com:6969/announce,&tr=udp://open.stealth.si:80/announce
```

Reply:
fixed with 5d752718a039497a458a4ba40e252a67dbd4b3fe

Reply:
@garfield69 nice catch. take a look at `emtrek`. it's still building the magnet. good work.

Reply:
@garfield69 take a look at `linuxtracker` too. It has errors.

Reply:
re: emtrek, see https://github.com/Jackett/Jackett/pull/10091#issuecomment-721463189

Reply:
re: linuxtracker, fixed in 47e4d9440942a7407fae52eae55c103d7d60cac5